response_operations_ui/controllers/cir_controller.py

```diff
@@ -0,0 +1,42 @@
+import json
+import logging
+
+import requests
+from flask import current_app as app
+from structlog import wrap_logger
+
+from response_operations_ui.common.credentials import fetch_and_apply_oidc_credentials
+from response_operations_ui.exceptions.exceptions import ExternalApiError
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+def get_cir_service_status():
+    session = requests.Session()
+    fetch_and_apply_oidc_credentials(session=session, client_id=app.config["CIR_OAUTH2_CLIENT_ID"])
+
+    logger.debug("Get CIR service status", session_headers=str(session.headers))
+
+    try:
+        response = session.get(app.config["CIR_API_URL"] + "/status")
+    except requests.ConnectionError as e:
+        logger.error("CIR service connection error", error=str(e))
+        raise ExternalApiError(None, "CIR0001: CIR service connection error: " + str(e))
```

Comment:
The `error_code` parameter can be replaced with an Enum from an `APiErrorCodes` class rather than this string literal.

The string `str(e)` concatenation is purely for visibility during initial development and integration with the CIR service and wouldn't normally be included.